lightningd/peer_control.c

```diff
@@ -1403,7 +1407,7 @@ void peer_connected(struct lightningd *ld, const u8 *msg)
 	peer = peer_by_id(ld, &id);
 	if (!peer)
 		peer = new_peer(ld, 0, &id, &hook_payload->addr,
-				hook_payload->incoming);
+				NULL, hook_payload->incoming);
```

Comment:
... use their_features here.

And 

```
...
else 
    peer_updated_features(peer, their_features).
```

Though frankly that function is only called once and is going to be clearer open-coded IMHO.

Reply:
did it in the open, removed `peer_update_features`